File: 0456-132-pattern/0456-132-pattern.java
class Solution {
    public boolean find132pattern(int[] nums) {
        Stack<Integer> stk= new Stack<>();
        int second=Integer.MIN_VALUE;
        for(int i=nums.length-1;i>=0;i--){
            if(nums[i]<second) return  true;
            while(!stk.isEmpty() && nums[i]>stk.peek()) second=stk.pop();
            stk.push(nums[i]);
        }
        return false;
    }
}

File: 0456-132-pattern/0456-132-pattern.py
class Solution:
    def find132pattern(self, nums: List[int]) -> bool:
        stack=[]
        second=float('-inf')
        for i in range(len(nums)-1,-1,-1):
            num=nums[i]
            if(num<second):
                return True
            while(len(stack)>0 and num>stack[-1]):
                second=stack.pop()
            stack.append(num)
        return False

            


